core/src/block.rs

```diff
@@ -32,6 +31,20 @@ fn cal_witnesses_root(vec: &[Transaction]) -> H256 {
     merkle_root(&witnesses[..])
 }
 
+pub(crate) fn cal_proposals_hash(vec: &[ProposalShortId]) -> H256 {
+    if vec.is_empty() {
+        H256::zero()
+    } else {
+        let mut bytes = Vec::with_capacity(vec.len() * 10);
+        for id in vec.iter() {
+            bytes
+                .write_all(&(&id as &[u8; 10])[..])
+                .expect(VEC_WRITE_ALL_EXPECT);
+        }
+        blake2b_256(bytes).into()
```

Comment:
It's better to use `Blake2b::update` on the object created by `hash::new_blake2b` instead of constructing the vector first then hashing it.

Reply:
I rebase from the latest develop branch (to sync code of `Script::hash()`), replace two `blake2b_256` to `new_blake2b`, then force-push.